great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,282 @@
+import copy
+import logging
+from typing import Any, Dict, List, Optional
+
+from great_expectations.data_context.types.base import BaseYamlConfig
+from great_expectations.marshmallow__shade import INCLUDE, Schema, fields, post_load
+from great_expectations.marshmallow__shade.decorators import post_dump
+from great_expectations.types import DictDot
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class DomainBuilderConfig(DictDot):
+    def __init__(
+        self,
+        class_name: str,
+        module_name: Optional[str] = None,
+        batch_request: Optional[Dict[str, Any]] = None,
+        **kwargs,
+    ):
+        self._class_name = class_name
+        self._module_name = module_name
+        self.batch_request = batch_request
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+            logger.info(
+                'Set "self.%s = %s" during DomainBuilderConfig instantiation', k, v
+            )
+
+    @property
+    def class_name(self) -> str:
+        return self._class_name
+
+    @property
+    def module_name(self) -> Optional[str]:
+        return self._module_name
+
+
+class DomainBuilderConfigSchema(Schema):
+    class Meta:
+        unknown = INCLUDE
+
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.domain_builder",
+    )
+    batch_request = fields.Dict(keys=fields.String(), required=False, allow_none=True)
+
+    @post_load
+    def make_config(self, data: dict, **kwargs) -> DomainBuilderConfig:
+        return DomainBuilderConfig(**data)
+
+    @post_dump
+    def remove_nulls(self, data: dict, **kwargs) -> dict:
+        res = copy.deepcopy(data)
+        for k, v in data.items():
+            if v is None:
+                res.pop(k)
+                logger.info("Removed '%s' due to null value", k)
+        return res
+
+
+class ParameterBuilderConfig(DictDot):
+    def __init__(
+        self,
+        parameter_name: str,
```

Comment:
@cdkini I would like to recommend a top-bottom change here -- instread of `parameter_name` use `name` to be aligned with the other configuration objects.  I would be happy to discuss to insure we do the right thing.  There is some context about the names under the overused "parameter" that causes this potential confusion (it should not).  Thanks!

Reply:
okay! I believe this adheres to the spec but if you think it makes more sense to change, I'm happy to do so.

Reply:
@cdkini Correct -- I would like to alter the spec for "a better spec" now that we are build it.  Of course this will be a global change -- so happy to discuss.

Reply:
Changed name! Resolving